### ISSUE TITLE
Having problem from creating new dashboard without having a chart associated with

### ISSUE THREAD

**Issue by kevinwen2k** [apache/superset#16514]
Here’s what I did:

1. Click the link on the first 2 screenshots
2. I got the 3rd screenshot with uri /superset/dashboard/8/.
3. Expand the See more on the right side, I got the 4th screenshot.
4. When I go to /dashboard/list/, I can see the broken dashboard on the top as 5th screenshot.

Apparently, the new dashboard was created already, when I clicked both links in the first 2 screenshots. The problem comes from the API call /api/v1/dashboard/8/datasets due to dashboard created without charts as the 6th screenshot.
The root cause of this error is the dashboard.datasources is an empty set throwing an exception at line 117 in file superset/dashboards/dao.py in the 7th screenshot. The following is the tracing code:

```
get_dashboard_and_datasets_changed_on, dao.py:116
<lambda>, api.py:279
wrapper, cache.py:181
dispatch_request, app.py:1936
full_dispatch_request, app.py:1950
wsgi_app, app.py:2447
__call__, app.py:2464
__call__, cli.py:337
debug_application, __init__.py:304
execute, serving.py:314
run_wsgi, serving.py:323
handle_one_request, serving.py:379
handle, server.py:427
handle, serving.py:345
__init__, socketserver.py:747
finish_request, socketserver.py:360
process_request_thread, socketserver.py:683
run, threading.py:870
_bootstrap_inner, threading.py:932
_bootstrap, threading.py:890
```
The same problem happens at line 94 on the same file (dao.py) when another api /api/v1/dashboard/8/charts is hit without charts on the dashboard shown as the 8th screenshot.
To successfully create a new dashboard, I have to create a chart and then create a new dashboard from there as the 9th screenshot.
I’m using superset 1.2 version. Does anyone have the same issue? Thanks.
![Screen Shot 2021-08-30 at 4 05 45 PM](https://user-images.githubusercontent.com/4207782/131419181-52511255-1f17-458c-921d-e2cc2bfe9bd7.png)
![Screen Shot 2021-08-30 at 4 05 35 PM](https://user-images.githubusercontent.com/4207782/131419195-17446368-6e37-49a7-9928-abb921cd1c89.png)
![Screen Shot 2021-08-30 at 4 10 44 PM](https://user-images.githubusercontent.com/4207782/131419226-d99dc91d-a264-4fe4-87ed-c3656620f720.png)
![Screen Shot 2021-08-30 at 4 07 53 PM](https://user-images.githubusercontent.com/4207782/131419231-b1319c1b-d39c-44cd-b1d7-491209d8b43b.png)
![Screen Shot 2021-08-30 at 4 16 38 PM](https://user-images.githubusercontent.com/4207782/131419241-49cb0b04-0e31-4daf-9d2f-712e8a073196.png)
![Screen Shot 2021-08-30 at 4 21 30 PM](https://user-images.githubusercontent.com/4207782/131419252-03eaf069-0800-49fa-a57f-b0a2bceb8036.png)
![Screen Shot 2021-08-30 at 4 27 00 PM](https://user-images.githubusercontent.com/4207782/131419259-620b7614-0e2c-4c92-8ca7-0b70f4358f7d.png)
![Screen Shot 2021-08-30 at 4 31 32 PM](https://user-images.githubusercontent.com/4207782/131419273-a2e29941-0aaf-4fe6-b261-3ec259e6386a.png)

**Comment by nytai** [comment]
The seems to be triggered by this call https://github.com/apache/superset/blob/e2469162fa074af229ff97e845755caf1a29ca0d/superset/dashboards/api.py#L225

When `DASHBOARD_CACHE` feature flag is enabled. 

cc @etr2460

**Comment by nytai** [comment]
Nevermind, looks like this was fixed by https://github.com/apache/superset/pull/14450

**Comment by nytai** [comment]
This should be fixed in v1.3
